### DISCUSSION TITLE
What is the purpose of the empty setup.py?

### DISCUSSION THREAD

**Discussion by jhossbach** [python-rope/rope#511] [Q&A] [answered]
As of the latest big merge, the setup.py is empty. I believe this to break some of the packaging repos, e.g. arch (https://archlinux.org/packages/community/any/python-rope/) as they try to install rope using `python setup.py install --skip-build`

**Comment by lieryan** [comment, marked as answer]
The empty setup.py is currently [needed for setuptools/pip to do an editable install](https://github.com/pypa/setuptools/blob/9c7163bc6030e546ba1133de823d0b4ac06a02d9/docs/userguide/pyproject_config.rst#configuring-setuptools-using-pyprojecttoml-files). Once this is fixed in pip, then we might eventually be able to remove it and rely only on pyproject.toml but from my understanding this is still necessary.

I'm not quite sure what the problem is with arch build or why they are using --skip-build, as I don't manage the arch package.

I think @Foxboron is the Arch maintainer for the rope package in Arch, would you be able to provide some context here?

**Comment by jhossbach** [comment]
I got here after getting a `ModuleNotFoundError`. After digging the [arch python package guidelines](https://wiki.archlinux.org/title/Python_package_guidelines) I found that `python-pytoolconfig` is not in the depends list in the `PKGBUILD`. Also, according to the guidelines the `python setup.py install ...` is used when the `pyproject.toml` does not exist or it fails to contain the `[build-system]` table, both of which are not true.
So my guess is that
 - `python-pytoolconfig` has to be added to the depends list and
 - the install method can be adapted to be PEP517 compliant

**Reply by jhossbach** [reply]
TL;DR I think is is a bug in the arch package

**Reply by jhossbach** [reply]
Just realized that a bug has already been reported.

**Reply by Foxboron** [reply]
I've been very lazy recently :)

**Reply by jhossbach** [reply]
Not a problem :) What about the second item though?

**Reply by Foxboron** [reply]
That's an issue for when a new `rope` release is made. The latest still works with `setup.py`.

**Reply by jhossbach** [reply]
Alright, thanks for the clearing up. :+1:
